lang/java/reef-common/src/main/java/org/apache/reef/driver/restart/DriverRestartManager.java

```diff
@@ -107,7 +108,16 @@ public synchronized void onRestart() {
 
     driverRuntimeRestartManager.informAboutEvaluatorFailures(getFailedEvaluators());
 
-    // TODO[REEF-560]: Call onDriverRestartCompleted() on a Timer.
+    if (driverRestartEvaluatorRecoveryMillis != Integer.MAX_VALUE) {
+      // Don't use Clock here because if there is an event scheduled, the driver will not be idle, even if
+      // driver restart has already completed, and we cannot cancel the event.
+      new Timer().schedule(new TimerTask() {
+        @Override
+        public void run() {
+          onDriverRestartCompleted();
```

Comment:
> Orthogonal to this PR, I think there are a few more ideas worth considering: Should we report the Evaluators to expect in DriverRestartedHandler, and should we have a special EvaluatorFailedHandler for failures on Driver restart s.t. the user can know the difference?

Maybe. If the the events are fired between DriverRestarted and RestartedComplete, the app could infer this anyhow, right?